### ISSUE TITLE
Cannot run midas_r_simple without additional low frequency variables

### ISSUE THREAD

**Issue by sweetmoniker** [mpiktas/midasr#55]
The documentation suggests this is possible (z=NULL), but I cannot get it to run. I always get the following error:
Error in matrix(z, ncol = 1) : 
  'data' must be of a vector type, was 'NULL'
Or am I not understanding the documentation correctly?

**Comment by vzemlys** [comment]
Yes this is a bug.

**Comment by vzemlys** [comment]
I've fixed in the development version of package. Usually you would not want to use `z=NULL`, since intercept is the low frequency variable and you would not want to fit a regression model without an the intercept.

**Comment by sweetmoniker** [comment]
Now when I pull the z variable from the call I get the following error:

Error in optimx.check(par, optcfg$ufn, optcfg$ugr, optcfg$uhess, lower,  : 
  Cannot evaluate function at initial parameters

Something about the way the arguments are being passed to the optimx function is failing. I tried to recreate the function manually and received the same result, but I'm not quite sure what some of the calls mean, specifically "all_coef." I added "weight = nealmon" to my call and the returned value basically said there was nothing to optimize. Sorry I can't be more helpful.
